src/sentry/discover/compare_tables.py

```diff
@@ -0,0 +1,193 @@
+    eap_data_row = snql_eap_result["data"][0] if len(snql_eap_result["data"]) > 0 else {}
+    metrics_data_row = (
+        metrics_query_result["data"][0] if len(metrics_query_result["data"]) > 0 else {}
+    )
+    mismatches = []
+    for field, data in metrics_data_row.items():
+        if is_equation(field):
+            continue
+        translated_field, *rest = translate_columns([field])
+        if data is not None and eap_data_row[translated_field] is None:
+            logger.info("Field %s not found in EAP response", field)
+            sentry_sdk.set_tag("dashboard_comparison_mismatch_field", field)
+            mismatches.append(field)
+
+    return (len(mismatches) == 0, mismatches)
+
+
+def compare_tables_for_dashboard_widget_queries(
+    widget_query: DashboardWidgetQuery,
+) -> CompareTableResultDict:
+    widget: DashboardWidget = widget_query.widget
+    dashboard: Dashboard = widget.dashboard
+    organization: Organization = dashboard.organization
+    projects: list[Project] = list(dashboard.projects.all())
+    if len(list(projects)) == 0:
+        return {
+            "passed": False,
+            "reason": CompareTableResult.NO_PROJECT,
+            "fields": None,
+            "widget_query": widget_query,
+            "mismatches": None,
+        }
+
+    project = projects[0]
+
+    fields = widget_query.fields
+    if len(fields) == 0:
+        return {
+            "passed": False,
+            "reason": CompareTableResult.NO_FIELDS,
+            "fields": None,
+            "widget_query": widget_query,
+            "mismatches": None,
+        }
+
+    snuba_dataclass = _get_snuba_dataclass_for_dashboard_widget(widget, list(projects))
+    selected_columns = _get_field_list(fields)
+    equations = [equation for equation in _get_equation_list(widget_query.fields or []) if equation]
+    query = widget_query.conditions
+
+    has_metrics_error = False
+    has_snql_eap_error = False
+
+    try:
+        metrics_query_result = metrics_query(
+            selected_columns,
+            query,
+            snuba_dataclass,
+            equations,
+            orderby=None,
+            offset=None,
+            limit=1,
+            referrer="dashboards.transactions_spans_comparison",
+            transform_alias_to_input_format=True,
+        )
+
+    except Exception as e:
+        logger.info("Metrics query failed: %s", e)
+        has_metrics_error = True
+
+    environments = dashboard.filters.get("environment", []) if dashboard.filters is not None else []
+
+    snuba_params = SnubaParams(
+        environments=environments,
+        projects=[project],
+        organization=organization,
+        stats_period="7d",
+    )
+
+    eap_query_parts = translate_mep_to_eap(
+        QueryParts(
+            query=query,
+            selected_columns=selected_columns,
+            equations=equations,
+            orderby=None,
+        )
+    )
+
+    try:
+        snql_eap_result = spans_rpc.run_table_query(
```

Comment:
nit: We don't construct snql for eap queries in sentry, we make RPCs, so let's rename
```suggestion
        eap_result = spans_rpc.run_table_query(
```